src/main/webapp/app/quiz/edit/drag-and-drop-question/edit-drag-and-drop-question.component.js

```diff
@@ -0,0 +1,644 @@
+
+    function togglePreview() {
+        vm.showPreview = !vm.showPreview;
+    }
+
+    // set up editor
+    vm.random = pseudoRandomLong();
+    var editor;
+    requestAnimationFrame(function () {
+        editor = ace.edit("question-content-editor-" + vm.random);
+        editor.setTheme("ace/theme/chrome");
+        editor.getSession().setMode("ace/mode/markdown");
+        editor.renderer.setShowGutter(false);
+        editor.renderer.setPadding(10);
+        editor.renderer.setScrollMargin(8, 8);
+        editor.setHighlightActiveLine(false);
+        editor.setShowPrintMargin(false);
+
+        generateMarkdown();
+
+        editor.on("blur", function () {
+            parseMarkdown(editor.getValue());
+            vm.onUpdated();
+            $scope.$apply();
+        });
+    });
+
+    /**
+     * generate the markdown text for this question
+     *
+     * The markdown is generated according to these rules:
+     *
+     * 1. First the question text is inserted
+     * 2. If hint and/or explanation exist, they are added after the text with a linebreak and tab in front of them
+     *
+     */
+    function generateMarkdown() {
+        var markdownText = (
+            vm.question.text +
+            (vm.question.hint ? "\n\t[-h] " + vm.question.hint : "") +
+            (vm.question.explanation ? "\n\t[-e] " + vm.question.explanation : "")
+        );
+        editor.setValue(markdownText);
+        editor.clearSelection();
+    }
+
+    /**
+     * Parse the markdown and apply the result to the question's data
+     *
+     * The question text is split at [-h] and [-e] tags.
+     *  => First part is text. Everything after [-h] is Hint, anything after [-e] is explanation
+     *
+     * @param questionText {string} the markdown text to parse
+     */
+    function parseMarkdown(questionText) {
+        // split question into main text, hint and explanation
+        var questionTextParts = questionText.split(/\[\-e\]|\[\-h\]/g);
+        vm.question.text = questionTextParts[0].trim();
+        if (questionText.indexOf("[-h]") !== -1 && questionText.indexOf("[-e]") !== -1) {
+            if (questionText.indexOf("[-h]") < questionText.indexOf("[-e]")) {
+                vm.question.hint = questionTextParts[1].trim();
+                vm.question.explanation = questionTextParts[2].trim();
+            } else {
+                vm.question.hint = questionTextParts[2].trim();
+                vm.question.explanation = questionTextParts[1].trim();
+            }
+        } else if (questionText.indexOf("[-h]") !== -1) {
+            vm.question.hint = questionTextParts[1].trim();
+            vm.question.explanation = null;
+        } else if (questionText.indexOf("[-e]") !== -1) {
+            vm.question.hint = null;
+            vm.question.explanation = questionTextParts[1].trim();
+        } else {
+            vm.question.hint = null;
+            vm.question.explanation = null;
+        }
+    }
+
```

Comment:
***IMPORTANT*** The code above seems to be duplicated from edit-multiple-choice-question.component.js
Please avoid that and rather refactor this method into another java script file that you can invoke from two places. Please also talk to @MoritzIssig who also uses this code in a third place

Reply:
Should we also wait with this refactoring until the two Pull Requests are merged (Same as with QuizExerciseResource.java)?